api/[lang]/detail/index.ts

```diff
@@ -1,19 +1,128 @@
-import { VercelRequest, VercelResponse } from '@vercel/node'
-import { getLang } from '../../../utils/lang'
+import { VercelRequest, VercelResponse } from "@vercel/node";
+import { fetchHTML } from "../../../utils/fetch";
+import { getLang } from "../../../utils/lang";
+import cheerio from "cheerio";
+import { parse, stringify } from "qs";
+
+export default async (
+  request: VercelRequest,
+  response: VercelResponse
+): Promise<void> => {
+  const langConfig = getLang(String(request.query.lang));
+  const detailUrl = request.query.url;
+  const url = request.headers["x-forwarded-host"];
+  const proto = request.headers["x-forwarded-proto"];
+  const query: any = request.query.query;
+  const page: any = +request.query.page || 0;
+  const portionamount = request.query.portionamount;
+  const portionid = request.query.portionid;
+
+  if (!langConfig) {
+    response.json({ error: `${request.query.lang} are not supported` });
+    return;
+  }
 
-export default async (request: VercelRequest, response: VercelResponse): Promise<void> => {
-  const langConfig = getLang(String(request.query.lang))
-  const detailUrl = request.query.url
   if (!detailUrl) {
-    response.json({ error: 'please provide detailLink on search' })
-    return
+    response.json({ error: "please provide detailLink on search" });
+    return;
   }
   if (!langConfig) {
-    response.json({ error: `${request.query.lang} are not supported` })
-    return
+    response.json({ error: `${request.query.lang} are not supported` });
+    return;
+  }
+
+  let _url = langConfig.baseUrl + detailUrl;
+  console.log("ss", _url);
+  const html = await fetchHTML(_url, {
+    portionamount,
+    portionid,
+    pg: page,
+  });
+
+  const $ = cheerio.load(html);
+  //console.log("ss2", html);
+  let items: any = {};
+
+  $("div.nutrient.left").each((_, elem: any) => {
+    const element = $(elem);
+    const normalizeText = element.text().replace(/(\r\n|\n|\r\t|\t|\r)/gm, "");
+    let item;
+    const nextELem = element.next().text();
+    console.log("normali", normalizeText);
+
+    switch (normalizeText) {
+      case "Enerji":
```

Comment:
I think this detail is too specific with one language. Can you make it more language agnostic? Maybe add some parameters on `utils/lang.ts`. Just maybe add this types and update accordingly.

```ts
detailTextSeparator: {
  energy: string;
  protein: string;
  fat: string
  ... etc
}
```